src/VisualStudio/IntegrationTest/TestUtilities/InProcess2/ChangeSignatureDialog_InProc2.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls.Primitives;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ChangeSignature;
+using Microsoft.VisualStudio.Threading;
+using Xunit;
+
+namespace Microsoft.VisualStudio.IntegrationTest.Utilities.InProcess2
+{
+    public class ChangeSignatureDialog_InProc2 : InProcComponent2
+    {
+        public ChangeSignatureDialog_InProc2(JoinableTaskFactory joinableTaskFactory, Editor_InProc2 editor)
+            : base(joinableTaskFactory)
+        {
+            Editor = editor;
+        }
+
+        private Editor_InProc2 Editor
+        {
+            get;
+        }
+
+        internal async Task<ChangeSignatureDialog> VerifyOpenAsync()
+        {
+            while (true)
+            {
+                var window = await TryGetDialogAsync();
+                if (window is null)
+                {
+                    await Task.Yield();
+                    continue;
+                }
+
+                await WaitForApplicationIdleAsync();
+                return window;
+            }
+        }
+
+        internal async Task VerifyClosedAsync()
+        {
+            while (true)
+            {
+                var window = Application.Current.Windows.OfType<ChangeSignatureDialog>().SingleOrDefault();
+                if (window is null)
+                {
+                    return;
+                }
+
+                await Task.Yield();
+            }
+        }
+
+        internal async Task<ChangeSignatureDialog> GetDialogAsync()
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync();
+            return Application.Current.Windows.OfType<ChangeSignatureDialog>().Single();
+        }
+
+        internal async Task<ChangeSignatureDialog> TryGetDialogAsync()
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync();
+            return Application.Current.Windows.OfType<ChangeSignatureDialog>().SingleOrDefault();
+        }
+
+        public async Task<bool> CloseWindowAsync()
+        {
+            if (await TryGetDialogAsync() is null)
+            {
+                return false;
+            }
+
+            await ClickCancelAsync();
+            return true;
+        }
+
+        public async Task InvokeAsync()
+        {
+            await Task.Yield();
+            await ExecuteCommandAsync("Refactor.RemoveParameters");
```

Comment:
😟 This is executed using <kbd>Ctrl</kbd>+<kbd>R</kbd>, <kbd>Ctrl</kbd>+<kbd>V</kbd> in the old implementation. However, the new implementation of SendKeys (based on `PostMessage` instead of `SendInput`) is not registering the <kbd>Ctrl</kbd> keys. The use of `ExecuteCommandAsync` holds us over (and isn't dependent on key bindings), but we should still figure out why we can't test with true keyboard input.

Reply:
The old implementation of `SendKeys` was based on `SendInput` because that is the correct/recommended way to simulate input.

PostMessage has numerous problems (such as special keys, `Ctrl`, `Shift`, `Alt`, etc, not working).

There are a number of articles/blogs around the web discussing the differences between the two (some from Microsoft: https://blogs.msdn.microsoft.com/oldnewthing/20050530-11/?p=35513).

Reply:
Yep, was already in the process of moving back to `SendInput`. `SendInput` causes severe problems for users attempting to run integration tests locally, so I was hoping to find another solution.

Reply:
➡️ Now removed.